### PR TITLE
CentOS 8 Support

### PR DESCRIPTION
**Pull request by yachub**
Please see the changes in this PR to add CentOS 8 for all templates per #9.

### CODE REVIEW

File: centos-8/http/ks.cfg [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+-iwl5150-firmware
+-iwl6000-firmware
+-iwl6000g2a-firmware
+-iwl6000g2b-firmware
+-iwl6050-firmware
+-iwl7260-firmware
+%end
+
+%post
+/usr/bin/yum -y install sudo
```
Comment (genebean) [inline centos-8/http/ks.cfg]: Is `yum` still what we should be calling here or does it need to be `dnf`?

File: centos-8/http/ks.cfg [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+-iwl3160-firmware
+-iwl3945-firmware
+-iwl4965-firmware
+-iwl5000-firmware
+-iwl5150-firmware
+-iwl6000-firmware
+-iwl6000g2a-firmware
+-iwl6000g2b-firmware
+-iwl6050-firmware
+-iwl7260-firmware
```
Comment (genebean) [inline centos-8/http/ks.cfg]: Will you check the initial package set and make sure this still omits all the firmwares that are not needed and doesn't try to omit things that no longer exist?

File: centos-8/scripts/sshd_config.sh [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+# Example of overriding settings on a per-user basis
+#Match User anoncvs
+#	X11Forwarding no
+#	AllowTcpForwarding no
+#	PermitTTY no
+#	ForceCommand cvs server
+
+AddressFamily inet
+EOF
```
Comment (genebean) [inline centos-8/scripts/sshd_config.sh:153]: I am pretty sure the original of this was the stock cent 7 file - will you verify it lines up with the stock cent 8 file?

File: centos-8/template-base-vars.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "os": "centos-8",
+  "iso_url": "http://ftp.osuosl.org/pub/centos/8/isos/x86_64/CentOS-8.2.2004-x86_64-minimal.iso",
+  "iso_checksum": "47ab14778c823acae2ee6d365d76a9aed3f95bb8d0add23a06536b58bb5293c0",
+  "iso_checksum_type": "sha256",
+  "boot_command_start": "<tab><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><bs><wait5>",
+  "boot_command_options": " biosdevname=0 net.ifnames=0 quiet vga=791",
+  "common_scripts": "../centos-common/scripts",
+  "vagrant_guest_os_type": "RedHat_64",
+  "vmware_guest_os_type": "centos8-64",
```
Comment (genebean) [inline centos-8/template-base-vars.json:10]: Have you validated these two are the needed ones?

File: centos-8/template-base-vars.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  "vmware_hardware_version": "14"
```
Comment (genebean) [inline centos-8/template-base-vars.json:11]: Have you validated that 14 includes support for cent 8?

File: centos-8/box-versions [hunk 6]
```diff
@@ -0,0 +1 @@
+nocm docker-ce puppet-latest puppet5 rvm-multi
```
Comment (genebean) [inline centos-8/box-versions:1]: does `docker-ce` even have a viable option on cent 8?

File: centos-8/http/ks.cfg (hunk 1 above)
Comment (yachub) [reply centos-8/http/ks.cfg]: `yum` still works, but might as well go ahead and change it to `dnf`.

File: centos-8/http/ks.cfg (hunk 2 above)
Comment (yachub) [reply centos-8/http/ks.cfg]: I removed one that does not exist `aic94xx-firmware`, and added these:

- crystalhd-firmware:  Firmware for the Broadcom Crystal HD video decoder
- iscan-firmware: Firmware for Epson flatbed scanners
- libertas-sd8686-firmware: Firmware for Marvell Libertas SD 8686 Network Adapter
- libertas-sd8787-firmware: Firmware for Marvell Libertas SD 8787 Network Adapter
- libertas-usb8388-firmware: Firmware for Marvell Libertas USB 8388 Network Adapter
- libertas-usb8388-olpc-firmware: OLPC firmware for Marvell Libertas USB 8388 Network Adapter
- uhd-firmware: Firmware files for the Universal Hardware driver (UHD).

File: centos-8/scripts/sshd_config.sh (hunk 3 above)
Comment (yachub) [reply centos-8/scripts/sshd_config.sh:153]: I copied the stock centos 8 file, but it looks like the `AddressFamily inet` is not in the stock centos 7 or centos 8. Do we want to keep that so ssh does not attempt to use ipv6?

File: centos-8/template-base-vars.json (hunk 4 above)
Comment (yachub) [reply centos-8/template-base-vars.json:10]: Yes according to `VBoxManage list ostypes` and the VMware vmx file these are correct.

https://www.packer.io/docs/builders/virtualbox/iso#guest_os_type
https://www.packer.io/docs/builders/vmware/iso#guest_os_type

File: centos-8/template-base-vars.json (hunk 5 above)
Comment (yachub) [reply centos-8/template-base-vars.json:11]: According to the [compatibility guide](https://www.vmware.com/resources/compatibility/search.php?deviceCategory=software&details=1&partner=272&operatingSystems=262&productNames=4&page=1&display_interval=10&sortColumn=Partner&sortOrder=Asc&testConfig=16)CentOS 8 and RHEL 8 are supported in Fusion 11.5 and 12, and ESXi 6.5 and 6.7 with hardware version 14

https://kb.vmware.com/s/article/67621

File: centos-8/box-versions (hunk 6 above)
Comment (yachub) [reply centos-8/box-versions:1]: Yes, there are repos for docker-ce on centos 8:

https://download.docker.com/linux/centos/8/x86_64/stable/Packages/

File: centos-8/scripts/sshd_config.sh (hunk 3 above)
Comment (genebean) [reply centos-8/scripts/sshd_config.sh:153]: No, let's remove that because we no longer disable ipv6.

File: centos-8/box-versions (hunk 6 above)
Comment (genebean) [reply centos-8/box-versions:1]: Cool - don't suppose you want to add one for podman, do you?

File: centos-8/scripts/sshd_config.sh (hunk 3 above)
Comment (yachub) [reply centos-8/scripts/sshd_config.sh:153]: Sounds good, it's already removed in the stock config then.

**Comment by genebean** [comment]
I finally got my vagrant and fusion stuff set back up so I can validate this today or this weekend. Sorry for the delay @thespain!

**Pull request merged**